File: python/242_有效的字母异位词_easy.py
# 使用哈希表，字母异位词一定是字符种类和数量相同的词
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        ans1, ans2 = {}, {}
        for ch1, ch2 in zip(s, t):
            if ch1 not in ans1:
                ans1[ch1]= 1
            else: ans1[ch1] += 1
            
            if ch2 not in ans2:
                ans2[ch2] = 1
            else: ans2[ch2] += 1
        return ans1 == ans2

File: cpp/242_有效的字母异位词_easy.cpp
// 使用哈希表，字母异位词一定是字符种类和数量相同的词

class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length())
            return false;
        unordered_map <char, int> ans1, ans2;
        for(int i=0; i<s.length(); i++){
            if(ans1.find(s[i]) == ans1.end())
                ans1[s[i]]=1;
            else ans1[s[i]]++;
            if (ans2.find(t[i]) == ans2.end())
                ans2[t[i]]=1;
            else ans2[t[i]]++;
        }
        return ans1 == ans2;
    }
};